src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -81,6 +81,7 @@ def __init__(  # pylint: disable=too-many-arguments
         self._channel: Optional[grpc.Channel] = None
         self.node = Node(node_id=SUPERLINK_NODE_ID)
         self._retry_invoker = _make_simple_grpc_retry_invoker()
+        self._message_ids: list[str] = []
```

Comment:
Since the message IDs must be unique, I think it makes sense to use `set` instead.
```suggestion
        self._message_ids: set[str] = set()
```